src/utils.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Returns whether or not the provided input is a JavaScript function.
+ *
+ * @param {any} input The input being tested.
+ *
+ * @return {boolean} Whether or not the provided input is a JavaScript function.
+ */
+export function isFunction(input) {
+  return typeof input === 'function';
+}
+
+/**
+ * Returns whether or not the provided input is a non-null JavaScript object.
+ *
+ * @param {any} input The input being tested.
+ *
+ * @return {boolean} Whether or not the provided input is a non-null JavaScript object.
+ */
+export function isNonNullObject(input) {
+  return Object.prototype.toString.call(input) === '[object Object]';
+}
+
+/**
+ * Returns whether or not the provided input is a Firebase Database Reference.
+ *
+ * @param {any} input The input being tested.
+ *
+ * @return {boolean} Whether or not the provided input is a Firebase Database Reference.
+ */
+export function isDatabaseReference(input) {
```

Comment:
Will be used in the next PR. If it's cool with you, I'll just keep it in this PR.